module/VuFind/tests/unit-tests/src/VuFindTest/Search/Solr/ResultsTest.php

```diff
@@ -124,6 +354,26 @@ protected function getResults(
         );
     }
 
+    /**
+     * Get Results objects from a response array
+     *
+     * @param array  $response Solr response array
+     * @param Params $params   Params
+     *
+     * @return Results
+     */
+    protected function getResultsFromResponse(
+        array $response,
+        Params $params
+    ): Results {
+        $collection = new RecordCollection($response);
+        $searchService = $this->createMock(SearchService::class);
+        $searchService->expects($this->once())
+            ->method('search')
```

Comment:
Is it worth validating the parameters sent to search using `->with()`? I imagine you could pass in an array of expectations if there's value in doing it and if you don't expect the input to be uniform across all tests.

Reply:
I'd say that not here, it's out of scope. E.g. `testGetResultTotal()` does its own validation, and since there are a lot of parameters, I think doing it with the helper method would become a bit messy.